### PR TITLE
Update EIP-1: Modernize section list and requirements

### PR DESCRIPTION
**Pull request by Pandapip1**
The section list was rather dense and hard to read. It was also particularly focused on Core EIPs, as evidenced by the wording not being significantly changed in 5 years.

There are a few normative changes to EIP-1:

- A limit of one section, and a requirement as to the order is added. This was always the case by editor consensus, but was never explicitly stated.
- The abstract now allows for a short discussion of the motivation, and also should define the scope of an EIP.
- The motivation section now explicitly states that readers should know that lack of interoperability is a problem implied when creating an EIP, and doesn't need to be included.
- The motivation section was made required for Core EIPs, since the former text stated that "A motivation section is critical for EIPs that want to change the Ethereum protocol."
- Guidelines are established for when another section conflicts with the Specification section, to avoid a repeat of EIP-712.
- The rationale section now requires authors to list any tradeoffs made, and allows authors to assume that the reader has read the motivation section.
- The backwards compatibility section was made non-optional, and placeholder text is suggested
- The backwards compatibility section has guidelines for post-final changes
- The test cases section now recommends inline key/value pairs over unit tests
- The reference implementation is now required to be minimal, and suggests authors submit more complete implementations to ercref and openzeppelin (mostly to make authors that like writing lots of code happy)
- The security considerations section now explicitly requires privacy considerations to be included, as well as providing a list of common security considerations to include.
- The security considerations section has guidelines for post-final changes
- Assets must be CC0 or CC0 multi-licensed, unless the author can't legally do so.

### CODE REVIEW

File: EIPS/eip-1.md [hunk 1]
```diff
@@ -95,20 +95,111 @@ A PR moving an EIP from Last Call to Final SHOULD contain no changes other than
...
+
+An EIP's preamble consists of RFC 822-style headers with EIP metadata. See [EIP Header/Preamble](#eip-header-preamble) for details.
+
+### Abstract
+
+**Required for all EIPs**
+
+The abstract is a short paragraph (multi-sentence) technical summary. This should be a very terse yet human-readable version of the specification and motivation sections, and should define the narrow scope of the EIP. Reading an abstract should be sufficient to generally understand the EIP.
+
+This is a normative section.
```
Comment (SamWilsn) [inline EIPS/eip-1.md:114]: The abstract is normative? I'm not sure I agree 100% on that, but I think I can see your reasoning.

**Review by SamWilsn — CHANGES_REQUESTED** [review]
Mixing content changes with formatting changes makes it hard to compare.

I love the new formatting, but I think each content change needs to be made in its own PR so we can discuss properly.

File: EIPS/eip-1.md (hunk 1 above)
Comment (Pandapip1) [reply EIPS/eip-1.md:114]: The abstract shouldn't change when the EIP turns final.

File: EIPS/eip-1.md [hunk 2]
```diff
@@ -95,20 +95,111 @@ A PR moving an EIP from Last Call to Final SHOULD contain no changes other than
...
+This is a normative section, but in the case of a conflict, the Specification section is considered more normative, and the Test Cases should be amended.
+
+### Reference Implementation
+
+**Optional for all EIPs**
+
+The Reference Implementation provides a minimal non-production-ready implementation that people can use to assist in understanding or implementing the specification. More complex, production-ready implementations can be submitted to [ERCRef](https://github.com/ercref/ercref-contracts) and/or [OpenZeppelin Contracts](https://github.com/OpenZeppelin/openzeppelin-contracts).
+
+This is a non-normative section. In the case of a conflict with the Specification section, the Reference Implementation should be amended.
+
```
Comment (gcolvin) [inline EIPS/eip-1.md:177]: In #7230 I've suggested that for Core EIPs this section requires at lease a link to the relevant commits to the Executable Specifications.

**Review by gcolvin — APPROVED** [review]

File: EIPS/eip-1.md (hunk 1 above)
Comment (gcolvin) [reply EIPS/eip-1.md:114]: @SamWilsn could you resolve this?  Git thinks you are requesting a change.  Or, @Pandapip1, could you make the chanbe.

I can see both arguments.  It does summarize the motivation, which is not normative.

**Review by Scamreno — COMMENTED** [review]
Pandapip1-modernize-section-list #7047

**Pull request closed without merging**